### PR TITLE
feat(PPDSC-2179): switch stories

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-2179](https://nidigitalsolutions.jira.com/browse/PPDSC-2179)

**What**

1. Background - changes to Switch storybook scenarios requested by design + slight refactoring
2. What did you do
- Simplified typing of Switch component and made sure checked prop passed down to icons (and added test)
- Added dynamic icon scenario to Storybook
- Visual changes to other Storybook scenarios

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on: 
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2179-switch-stories 
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2179-switch-stories/storybook

**Review by baburay23 — DISMISSED** [review]
looks good to me.

File: src/switch/__tests__/switch.stories.tsx [hunk 1]
```diff
@@ -179,6 +174,12 @@ export const StorySwitchFieldset = () => (
                 thumbIcon: IconFilledDragHandle,
                 onIcon: IconFilledCheck,
                 offIcon: IconFilledClose,
+                thumb: {
```
Comment (Xin00163) [inline src/switch/__tests__/switch.stories.tsx:177]: Sorry I just noticed there is a reduced motion scenario. The should be no transitionPreset when the user enabled `Reduce motion` in system preferences, so I don't think we need to have a scenario for it or manually remove the transitionPreset.

**Review by Xin00163 — COMMENTED** [review]
Just one comment about the reduced motion. The rest looks good.
I noticed a comment "// TODO: Contains similar logic to getComponentOverrides to cast the override icon as a component. But why is this necessary?"  Is there anything we need to do?

File: src/switch/__tests__/switch.stories.tsx (hunk 1 above)
Comment (mstuartf) [reply src/switch/__tests__/switch.stories.tsx:177]: I'll change the label for this scenario. It's only there to show how the Switch works with screen readers when part of a fieldset.

**Comment by mstuartf** [comment]
> Just one comment about the reduced motion. The rest looks good. I noticed a comment "// TODO: Contains similar logic to getComponentOverrides to cast the override icon as a component. But why is this necessary?" Is there anything we need to do?

@Xin00163 I've removed this comment.

**Review by Xin00163 — APPROVED** [review]

**Review by Xin00163 — DISMISSED** [review]

**Review by baburay23 — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**